3-networks/modules/restricted_shared_vpc/main.tf

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/******************************************
+  Shared VPC configuration
+ *****************************************/
+
+locals {
+  vpc_type                   = "shared"
+  vpc_label                  = "restricted"
+  vpc_name                   = "${var.environment_code}-${local.vpc_type}-${local.vpc_label}"
+  network_name               = "vpc-${local.vpc_name}"
+  restricted_googleapis_cidr = "199.36.153.4/30"
+}
+
+module "main" {
+  source                                 = "terraform-google-modules/network/google"
+  version                                = "~> 2.0"
+  project_id                             = var.project_id
+  network_name                           = local.network_name
+  shared_vpc_host                        = "true"
+  delete_default_internet_gateway_routes = "true"
+
+  subnets          = var.subnets
+  secondary_ranges = var.secondary_ranges
+
+  routes = concat(
+    [{
+      name              = "rt-${local.vpc_name}-1000-all-default-restricted-api"
+      description       = "Route through IGW to allow restricted google api access."
+      destination_range = local.restricted_googleapis_cidr
+      next_hop_internet = "true"
+      priority          = "1000"
+    }],
+    var.nat_enabled ?
+    [
+      {
+        name              = "rt-${local.vpc_name}-1000-egress-internet-default"
+        description       = "Tag based route through IGW to access internet"
+        destination_range = "0.0.0.0/0"
+        tags              = "egress-internet"
+        next_hop_internet = "true"
+        priority          = "1000"
+      },
+      {
+        name              = "rt-${local.vpc_name}-1000-all-default-windows-kms"
```

Comment:
nit: this probably should not be bundled with NAT as windows activation is separate. If I recall correctly, it can still work with Google private access enabled, without NAT.